### DISCUSSION TITLE
Electron and BrainpoolP

### DISCUSSION THREAD

**Discussion by KillerCodeMonkey** [panva/jose#226] [Q&A] [answered]
Heyho,

thanks for the great package!

For a project we need to use BrainpoolP256r1. The problem is, that node >=14.x is supporting those elliptic curves (via crypto openssl module), but electron bundles boringssl instead.

BoringSSL will not implement those curves, so we are using elliptic and defining this curve by ourself.

So we have our curve (registered via elliptic) and a point on that curve as public key (we get from an api).
With this key we need to create a JWE.

Do you know if it is somehow possible to extend supported curves and algs or register it with jose?
I know your project is using node crypto under the hood, as well, but maybe you have an idea how to achieve that?

We just do not want to create our own custom implementation of that :(.

Thanks for your time and patience.

**Comment by panva** [comment, marked as answer]
The library only supports iana registered algorithms and curves registered for use in JOSE. It also only supports native crypto runtimes.

**Reply by KillerCodeMonkey** [reply]
And just for interest: do you know if a custom implementation of that encryption part is even possible?

**Reply by panva** [reply]
ecdh-es is not used for the actual encryption, it is used to derive the shared secret that goes through ConcatKDF to end up with the actual used encryption key.

**Reply by KillerCodeMonkey** [reply]
thanks for your time :)
